packages/cw721/src/state.rs

```diff
@@ -0,0 +1,241 @@
+use std::marker::PhantomData;
+
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, BlockInfo, CustomMsg, Decimal, StdResult, Storage, Timestamp};
+use cw_ownable::{OwnershipStore, OWNERSHIP_KEY};
+use cw_storage_plus::{Index, IndexList, IndexedMap, Item, Map, MultiIndex};
+use cw_utils::Expiration;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+/// Creator owns this contract and can update collection info!
+/// !!! Important note here: !!!
+/// - creator is stored using using cw-ownable's OWNERSHIP singleton, so it is not stored here
+/// - in release v0.18.0 it was used for minter (which is confusing), but now it is used for creator
+pub const CREATOR: OwnershipStore = OwnershipStore::new(OWNERSHIP_KEY);
+/// - minter is stored in the contract storage using cw_ownable::OwnershipStore (same as for OWNERSHIP but with different key)
+pub const MINTER: OwnershipStore = OwnershipStore::new("collection_minter");
+
+/// Default CollectionInfoExtension with RoyaltyInfo
+pub type DefaultOptionCollectionInfoExtension = Option<CollectionInfoExtension<RoyaltyInfo>>;
+pub type DefaultOptionMetadataExtension = Option<Metadata>;
+
+pub struct Cw721Config<
+    'a,
+    TMetadata,
+    TCustomResponseMessage,
+    TExtensionExecuteMsg,
+    TMetadataResponse,
+    TCollectionInfoExtension,
+> where
+    TMetadata: Serialize + DeserializeOwned + Clone,
+    TMetadataResponse: CustomMsg,
+    TExtensionExecuteMsg: CustomMsg,
+    TCollectionInfoExtension: Serialize + DeserializeOwned + Clone,
+{
+    /// Note: do not use deprecated/legacy key "nft_info"!
+    pub collection_info: Item<'a, CollectionInfo<TCollectionInfoExtension>>,
+    pub token_count: Item<'a, u64>,
+    /// Stored as (granter, operator) giving operator full control over granter's account.
+    /// NOTE: granter is the owner, so operator has only control for NFTs owned by granter!
+    pub operators: Map<'a, (&'a Addr, &'a Addr), Expiration>,
+    /// Note: do not use deprecated/legacy keys "tokens" and "tokens__owner"!
+    pub nft_info: IndexedMap<'a, &'a str, NftInfo<TMetadata>, TokenIndexes<'a, TMetadata>>,
+    pub withdraw_address: Item<'a, String>,
```

Comment:
`Cw721Config` now stores all relevant data in one place (easier for contracts to use) - except for 2 external stores using `cw-ownable`: ...